packages/mgt-element/src/utils/LocalizationHelper.ts

```diff
@@ -0,0 +1,140 @@
+/**
+ * -------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the MIT License.
+ * See License in the project root for license information.
+ * -------------------------------------------------------------------------------------------
+ */
+
+import { EventDispatcher, EventHandler } from './EventDispatcher';
+
+/**
+ * Helper class for Localization
+ *
+ *
+ * @export
+ * @class LocalizationHelper
+ */
+export class LocalizationHelper {
+  static _strings: any;
+
+  static _stringsEventDispatcher: EventDispatcher<any> = new EventDispatcher();
+
+  static _directionEventDispatcher: EventDispatcher<any> = new EventDispatcher();
+
+  private static mutationObserver;
+
+  public static get strings() {
+    return this._strings;
+  }
+
+  /**
+   * Set strings to be localized
+   *
+   * @static
+   * @memberof LocalizationHelper
+   */
+  public static set strings(value: any) {
+    this._strings = value;
+    this._stringsEventDispatcher.fire(null);
+  }
+
+  /**
+   * returns body dir attribute to determine rtl or ltr
+   *
+   * @static
+   * @returns {string} dir
+   * @memberof LocalizationHelper
+   */
+  public static getDocumentDirection() {
+    this.initDirection();
+    return document.body.getAttribute('dir') || document.documentElement.getAttribute('dir');
+  }
+
+  /**
+   * Fires event when LocalizationHelper changes state
+   *
+   * @static
+   * @param {EventHandler<ProvidersChangedState>} event
+   * @memberof LocalizationHelper
+   */
+  public static onStringsUpdated(event: EventHandler<any>) {
+    this._stringsEventDispatcher.add(event);
+  }
+
+  public static removeOnStringsUpdated(event: EventHandler<any>) {
+    this._stringsEventDispatcher.remove(event);
+  }
+
+  public static onDirectionUpdated(event: EventHandler<any>) {
```

Comment:
Make sure to use this in baseComponent so the components know when the direction changes

Reply:
updated, interested to hear your thoughts on this:

https://github.com/microsoftgraph/microsoft-graph-toolkit/commit/c4cfc801fd9975636500ab3cee19ff2e618a9444#diff-cedee173f5b1bd643953206e4e1046d63137256f7753a05b9d6e451cd5b211f9R135

This line was added to compensate for components used within other components (e.g. mgt-person inside of mgt-login) otherwise the internalProperty `direction` isn't updated and will return the default "ltr".

Reply:
Makes sense to me - looks good